sys/trickle/trickle.c

```diff
@@ -33,25 +35,21 @@ void trickle_callback(trickle_t *trickle)
 
 void trickle_interval(trickle_t *trickle)
 {
-    uint32_t old_interval = trickle->I;
-    uint32_t max_interval = trickle->Imin << trickle->Imax;
-    uint32_t diff = old_interval - trickle->t;
+    assert(trickle->I > 0);
 
-    trickle->I *= 2;
+    uint32_t diff = trickle->I - trickle->t;
 
-    if ((trickle->I == 0) || (trickle->I > max_interval)) {
-        trickle->I = max_interval;
-    }
+    trickle->I *= 2;
+    trickle->I = MIN(trickle->I, (trickle->Imin << trickle->Imax));
```

Comment:
if it wraps over it's not such a big problem, as the operation will continue; I will further increase incrementally. However, in your solution, if I is 0, it will stay at 0 forever.